src/Dialogs/FingerprintDialog.vala

```diff
@@ -0,0 +1,163 @@
+        cancel_button.add_css_class (Granite.STYLE_CLASS_DESTRUCTIVE_ACTION);
+
+        try {
+            fp_utils = new SwitchboardPlugUserAccounts.FPUtils ();
+            finish_button = add_button (_("Finish"), Gtk.ResponseType.OK);
+            finish_button.add_css_class (Granite.STYLE_CLASS_SUGGESTED_ACTION);
+            finish_button.sensitive = false;
+        } catch (Error e) {
+            warning ("Failed to initialize fingerprint device: %s".printf (e.message));
+            status_label.label = _("Fingerprint device not available");
+            cancel_button.sensitive = true;
+            return;
+        }
+
+
+        response.connect ((response_id) => {
+            destroy ();
+        });
+
+        FPUtils.device.enroll_status.connect (set_progress);
+
+        this.unmap.connect (() => {
+            fp_utils.enroll_stop ();
+            fp_utils.release ();
+            FPUtils.device.enroll_status.disconnect (set_progress);
+        });
+
+        if (fp_utils.claim ()) {
+            fp_utils.enroll_start_async.begin ((obj, res) => {
+                bool success = fp_utils.enroll_start_async.end (res);
+                if (success) {
+                    progress_bar = new Gtk.LevelBar () {
+                        min_value = 1,
+                        max_value = FPUtils.device.num_enroll_stages + 1,
+                        value = 1,
+                        margin_top = 12,
+                        mode = DISCRETE
+                    };
+                    progress_revealer.child = progress_bar;
+                } else {
+                    destroy ();
+                }
+            });
+        }
+    }
+
+    private void set_progress (string message, bool done) {
+        status_label.label = get_friendly_status (message);
+        if (done) {
+            status_label.label = _("You are all set!");
+            cancel_button.sensitive = false;
+            finish_button.sensitive = true;
+            finish_button.grab_focus ();
+        }
+    }
+
+    private string get_friendly_status (string status) {
+        switch (status) {
+            case "enroll-completed":
+                progress_bar.value = progress_bar.max_value;
+                fingerprint_image.icon_name = "fingerprint-12-symbolic";
+                title_label.label = _("Enrollment Complete!");
+                return _("Enrollment complete");
+            case "enroll-failed":
+                fp_utils.enroll_stop ();
+                return _("Enrolling failed. Please try again");
+            case "enroll-stage-passed":
+                progress_revealer.reveal_child = true;
+                progress_bar.value = ++current_stage_count;
+                var progress = (progress_bar.value / progress_bar.max_value) * 12;
+                fingerprint_image.icon_name = "fingerprint-%d-symbolic".printf ((int) progress);
+                return _("Stage Passed! Carry on!");
```

Comment:
"Carry on" is very `en_GB` coded 😅 What action does "carrying on" entail? We should be explicit. Touch ID for example says "Lift and rest your finger on [the sensor] repeatedly"